opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchQueryTest.java

```diff
@@ -0,0 +1,204 @@
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("prefix_length", "0")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_fuzzy_transpositions_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("fuzzy_transpositions", "true")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_fuzzy_rewrite_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("fuzzy_rewrite", "constant_score")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_lenient_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("lenient", "false")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_operator_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("operator", "OR")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_minimum_should_match_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("minimum_should_match", "3")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_zero_terms_query_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("zero_terms_query", "NONE")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  @Test
+  public void test_boost_parameter() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "field_value"),
+        namedArgument("query", "query_value"),
+        namedArgument("boost", "1")
+    );
+    Assertions.assertNotNull(matchQuery.build(new MatchExpression(arguments)));
+  }
+
+  private class MatchExpression extends FunctionExpression {
+    public MatchExpression(List<Expression> arguments) {
+      super(MatchQueryTest.this.match, arguments);
+    }
+
+    @Override
+    public ExprValue valueOf(Environment<Expression, ExprValue> valueEnv) {
+      return null;
+    }
+
+    @Override
+    public ExprType type() {
```

Comment:
Why does type need to return null here?

Reply:
Function "type" needs implementation to fulfill "Expression" abstract class.